paimon-core/src/main/java/org/apache/paimon/table/source/DataSplit.java

```diff
@@ -141,6 +144,44 @@ public long mergedRowCount() {
         return partialMergedRowCount();
     }
 
+    public Object minValue(DataField dataField) {
+        Object minValue = null;
+        for (int i = 0; i < dataFiles().size(); i++) {
+            DataFileMeta dataFile = dataFiles.get(i);
+            Object other =
+                    InternalRowUtils.get(
+                            dataFile.valueStats().minValues(), dataField.id(), dataField.type());
```

Comment:
You should get valueStatsCol here, the index should be re-compute here by valueStatsCol.